dali/pipeline/data/tensor_vector.cc

```diff
@@ -62,11 +63,112 @@ TensorVector<Backend>::TensorVector(TensorVector<Backend> &&other) noexcept {
   other.tensors_.clear();
 }
 
-
 template <typename Backend>
-size_t TensorVector<Backend>::nbytes() const noexcept {
+void TensorVector<Backend>::UnsafeSetSample(int dst, const TensorVector<Backend> &owner, int src) {
+  // TODO(klecki): more consistency checks, contiguous -> non-contiguous removes shares_data from
+  // samples
+  if (type() == DALI_NO_TYPE && owner.type() != DALI_NO_TYPE) {
+    set_type(owner.type());
+  }
+  if (!order()) {
+    set_order(owner.order());
+  }
+  // Bounds check
+  assert(dst >= 0 && dst < static_cast<int>(curr_tensors_size_));
```

Comment:
Can we _please_ rename this into something that isn't misread quite as easily?
```suggestion
  assert(dst >= 0 && dst < static_cast<int>(curr_num_tensors_));
```
?
A bounds check should be self-explanatory... Also, `TensorVector` is the _only_ structure in our code base where `num_samples` is a `size_t`, not an `int`.

Reply:
37 occurrences in main branch, I don't want to inflate the number of lines in this PR. It can be done separately.